src/plugins/matrixops/insert.cpp

```diff
@@ -0,0 +1,362 @@
+//   Copyright (c) 2017 Shahrzad Shirzad
+//
+//   Distributed under the Boost Software License, Version 1.0. (See accompanying
+//   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/matrixops/insert.hpp>
+#include <phylanx/util/matrix_iterators.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+#include <hpx/util/optional.hpp>
+
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    match_pattern_type const insert::match_data = {match_pattern_type{"insert",
+        std::vector<std::string>{
+            "insert(_1, _2, _3)", "insert(_1, _2, _3, _4)"},
+        &create_insert, &create_primitive<insert>, R"(
+            arr, obj, values, axis
+            Args:
+
+                arr (array_like) : input array
+                obj (int, slice or sequence of ints) : Object that defines the index or
+                                                       indices before which values is
+                                                       inserted.
+                values (array_like) : Values to insert into arr.
+                axis (int, optional) : Axis along which to insert values. If axis is
+                                       None then arr is flattened first.
+
+            Returns:
+
+            A copy of arr with values inserted.
+            )",
+        true}};
+
+    ///////////////////////////////////////////////////////////////////////////
+    insert::insert(primitive_arguments_type&& operands, std::string const& name,
+        std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+      , dtype_(extract_dtype(name_))
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    template <typename T>
+    std::size_t insert::get_vec(ir::node_data<T> const& arg) const
```

Comment:
Would it be better to chose a more fitting name, like `get_vector_size` or similar? Additionally, this function could be `static` (but that's a very minor detail).